packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -392,40 +392,54 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
     private async summarizeInternal(fullTree: boolean, trackState: boolean): Promise<ISummarizeInternalResult> {
         await this.realize();
 
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        const summarizeResult = await this.channel!.summarize(fullTree, trackState);
-        let pathPartsForChildren: string[] | undefined;
+        // Create attributes to be added to the summary.
+        const { pkg, isRootDataStore } = await this.getInitialSnapshotDetails();
 
-        if (!this.disableIsolatedChannels) {
-            // Wrap dds summaries in .channels subtree.
-            wrapSummaryInChannelsTree(summarizeResult);
-            pathPartsForChildren = [channelsTreeName];
+        let summarizeResult;
+        try {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            summarizeResult = await this.channel!.summarize(fullTree, trackState);
         }
+        catch (error) {
+            // Log helpful information (currently the package path) if there is a summary failure.
+            this.logger.sendErrorEvent(
+                { eventName: "failToSummarize", fluidDataStoreId: this.id, packagePath: JSON.stringify(pkg) },
```

Comment:
`pkg` is PII. You cannot directly log it. I believe you will have to do something like this - https://github.com/microsoft/FluidFramework/blob/6e977e8e738fb68717317991615fc35375d6f04a/packages/runtime/container-runtime/src/dataStoreContext.ts#L263

Reply:
Will do!

Reply:
Can we do this as a performance event?  That will take care of the logging too.  So wrap everything in this function in `PerformanceEvent.timedExecAsync`.

Reply:
I am curious as to what's the advantage of logging this as a performance event? We don't really care about the time taken to summarize a data store. We already have the time to summarize the entire container and that's what we care about right.

Reply:
Also, we don't want to always log it, only when it fails. Otherwise this will be very noisy.